lib/route-recognizer.ts

```diff
@@ -290,26 +289,23 @@ class State {
 //  * prefers more static segments to more
 function sortSolutions(states: State[]) {
   return states.sort(function(a, b) {
-    if (!a.types) {
-      return b.types ? -1 : 0;
-    } else if (!b.types) {
-      return 1;
-    }
-    if (a.types.stars !== b.types.stars) { return a.types.stars - b.types.stars; }
+    let [ astatics, adynamics, astars ] = a.types || [0, 0, 0];
```

Comment:
Can you double check, that this destructuring output is reasonable. I know some transpiler modes, produce something that also enables iterators, which is far from ideal.